### PR TITLE
CH-142 Add dockerfile build args support to apps

### PR DESCRIPTION
**Pull request by condar-metacell**
Closes [CH-142](https://metacell.atlassian.net/browse/CH-142)

# Implemented solution

A `dockerfile` section has been added under the `haress` section of an application which then has a `buildArgs` map which is provided to the dockerfile by Skaffold when building.

# How to test this PR

Either run `pytest` to run the added unit test, or run `harness-deployment -i samples` and `skaffold dev` to see the value set in the values echoed out during build.

# Sanity checks:
- [x] The pull request is explicitly linked to the relevant issue(s)
- [x] The issue is well described: clearly states the problem and the general proposed solution(s)
- [x] In this PR it is explicitly stated how to test the current change
- [x] The labels in the issue set the scope and the type of issue (bug, feature, etc.)
- [ ] The relevant components are indicated in the issue (if any)
- [ ] All the automated test checks are passing
- [x] All the linked issues are included in one Sprint
- [x] All the linked issues are in the Review state
- [x] All the linked issues are assigned

# Breaking changes (select one):
- [x] The present changes do not change the preexisting api in any way
- [ ] This PR and the issue are tagged as a `breaking-change` and the migration procedure is well described [above](#implemented-solution)

# Possible deployment updates issues (select one):
- [ ] There is no reason why deployments based on CloudHarness may break after the current update
- [ ] This PR and the issue are tagged as `alert:deployment`

### Test coverage (select one):
- [ ] Tests for the relevant cases are included in this pr
- [ ] The changes included in this pr are out of the current test coverage scope

### Documentation (select one):
- [ ] The documentation has been updated to match the current changes
- [ ] The changes included in this PR are out of the current documentation scope

### Nice to have (if relevant):
- [ ] Screenshots of the changes
- [ ] Explanatory video/animated gif

[CH-142]: https://metacell.atlassian.net/browse/CH-142?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

**Review by zsinnema — COMMENTED** [review]
@condar-metacell I left a comment for @filippomc in the jira ticket (https://metacell.atlassian.net/browse/CH-142?focusedCommentId=15079)
I'm not sure we should implement this feature.

Even if this is only for Skaffold I'm not sure why we need this. Hope that Filippo can explain the need.

**Comment by filippomc** [comment]
@zsinnema this is useful for local development and debugging as mentioned in the Jira card, but I see your concern about this being abused as an attempt to set env variables on CI/CD and production systems. Let's be sure that the documentation states clearly that it's a development feature @condar-metacell

File: tools/deployment-cli-tools/ch_cli_tools/skaffold.py [hunk 1]
```diff
@@ -255,4 +275,14 @@ def get_image_tag(name):
     if not os.path.exists(os.path.dirname(vscode_launch_path)):
         os.makedirs(os.path.dirname(vscode_launch_path))
     with open(vscode_launch_path, 'w') as f:
-        json.dump(vs_conf, f, indent=2, sort_keys=True)
\ No newline at end of file
+        json.dump(vs_conf, f, indent=2, sort_keys=True)
+
+
+def get_additional_build_args(helm_values: dict, app_key: str) -> dict[str, str]:
```
Comment (filippomc) [inline tools/deployment-cli-tools/ch_cli_tools/skaffold.py:281]: the typing is wrong as helm_values is not a dict. It's HarnessMainConfig

**Review by filippomc — CHANGES_REQUESTED** [review]
Implementation looks good to me (only one minor remark on arg typing).

The main issue for me is the lack of documentation that gives clarity about this feature and possible misuses (see other comment). Another point to remark in the docs will be the fact that the feature is only supported in Skaffold and not in Codefresh CI/CD. Honestly I don't like much the incoherence although I agree with @zsinnema  that the use in CI/CD is not a use of this feature we want to promote.

**Review by filippomc — APPROVED** [review]
LGTM thanks @condar-metacell for the additional documentation.

**Pull request merged**